### ISSUE TITLE
Add lint.flake8-bandit setting to ignore specific argument names for S106 and S107

### ISSUE THREAD

**Issue by jamesdow21** [astral-sh/ruff#14365]
It'd be nice if there was a setting for `lint.flake8-bandit` to be able to ignore specific argument names from getting flagged as hard-coded passwords.

My use-case is that I have a `SecretsManager` class with a method signature like
```python
class SecretsManager:
    def get_secret_value(self, secret_id: str) -> Secret: ...
```

I created it to abstract out which secret management "backend" I was using when removing a ton of AWS-specific implementations and mostly stuck with the naming convention for `boto3.SecretsManager.Client` (which uses the argument name `SecretId`).

ruff will then end up flagging a S106 violation every time I retrieve any secret using `secret_id` as a keyword argument

Ideally, I could add a setting to my pyproject.toml like 
```toml
[tool.ruff.lint.flake8-bandit]
hardcoded-password-ignore = ["secret_id"]
```

I'm not sure how common an issue this is since I only see 2 potentially related issues ([here](https://github.com/astral-sh/ruff/issues/8789) and [here](https://github.com/astral-sh/ruff/issues/2384)) (and none in the original flake8-bandit repo), but it would at least be nice for me personally.

I know anything touching the security related rules is pretty sensitive though since a false positive is far less costly than a false negative

**Comment by MichaReiser** [comment]
Somewhat related: https://github.com/astral-sh/ruff/issues/8789

**Comment by bstrdsmkr** [comment]
Adding my voice to the pile, we want to catch cases where developers are testing using "real" credentials (to the test envs, but still...) and accidentally leave the credential in the test. Unfortunately, the test suites are the most likely place for developers to be iterating with secret material and forget to sanitize when it finally works

We also want to maintain the absolute highest signal-to-noise ratio possible, so we don't want to pop warnings for known-safe values.

For example, in a test, we might have:
```python
import os

def test_a_secret():
    os.environ["WEBHOOK_SECRET"] = "test-secret"
```
which pops a warning, but is pretty clearly safe so we want to suppress that warning, but want to catch something like this:
```python
import os

def test_a_secret():
    os.environ["WEBHOOK_SECRET"] = "tok_8675309"
```
The ideal solution for us would mirror the existing [dummy-variable-regex](https://docs.astral.sh/ruff/settings/#lint_dummy-variable-rgx) something like:
```toml
[tool.ruff.lint]
# Only ignore potential S105 secret violations prefixed with "test_".
safe-secret-rgx = "^test_.*$"
```
